---
title: Allele search
sitemap: sequence map
tools_menu: true
---

<H2>Allele Search</H2>
<form>
  <label class="uk-form-label uk-text-default uk-text-bold" for="collection-name">Step 1. Select a variant collection. Note that there are three reference assemblies to choose from.</label>
  <div>
    <select id="collection-name" class="uk-select uk-form-width-large uk-margin-small-top">
      {% assign genus = "Glycine" %}
      {% assign species = "max" %}
      {% for metadata in site.data.datastore-metadata[genus][species].diversity %}
        {% for metadata_file in metadata[1] %}
          {% if metadata_file[0] contains "MANIFEST" %}
            {% assign manifest = metadata_file %}
              {% for item in manifest[1] %}
              {% if item.applications contains "jbrowse" %}
                {% assign filename_parts = item.name | split: "." %}
                {% assign gensp = filename_parts[0] %}
                {% assign coll_parts = filename_parts | slice: 1, 4 %}
                {% assign diversity_base = filename_parts | slice: 1, 2 | join: "." %}
                {% capture collection_name %}{{ coll_parts| join: "." }}{% endcapture %}
                {%- capture collection_item -%}
                {
                  "name": "{{item.name}}",
                  "collection-name": "{{collection_name}}",
                  "description": "{{item.description}}"
                }
                {%- endcapture %}
                <option value='{{- collection_item | strip_newlines | strip_html -}}'>{{collection_name}}</option>
              {% endif %}
            {% endfor %}
          {% endif %}
        {% endfor %}
      {% endfor %}
    </select>
  </div>
  <div style="margin-left: 10px;">
    <p class="uk-text-default uk-text-emphasis" for="collection-name">Synopsis: <span id="synopsis"></span></p>
  </div>

  <div class="uk-margin" style="display: flex; justify-content: start;">
    <div>
      <div>
        <label class="uk-form-label uk-text-default uk-text-bold" for="gene-name">Step 2. Enter gene ID and size of left and right flanking regions...</label>
        <div><input class="uk-input uk-form-width-large uk-margin-small-top" type="text" id="gene-name" placeholder="Enter gene name"></input></div>
      </div>
      <div>
        <p class="uk-text-default uk-text-secondary" >e.g glyma.Wm82.gnm4.ann1.Glyma.16G044100</p>
      </div>
    </div>
      <div >
        <br>
        <div class="uk-margin-small-left">
          <input class="uk-input uk-form-width-medium uk-margin-small-top" type="text" id="flank-region"  placeholder="Enter flanking region">
        </input>
        <div>
          <label class="uk-form-label uk-text-default uk-text-secondary" for="flank-region">e.g 50000</label>
        </div>
      </div>
    </div>
  </div>
  
  <div class="uk-margin">
      <div>
          <label class="uk-form-label uk-text-default uk-text-bold" for="genomic-region">... or a genomic region, as a chromosome: start-end.</label>
      </div>
    <div>
      <input class="uk-input uk-form-width-medium uk-margin-small-top" type="text" id="genomic-region" rows="10" cols="30" placeholder="Start-End"></input>
      <div>
        <div><p class="uk-text-default uk-text-secondary">e.g 1-100000</p></div>
      </div>
    </div>
      <input class="uk-button uk-margin" uk-toggle="target: #modal" id="submit" type="button" value="Search" onclick="getData()" style="margin-top: 10px;"/> 
  </div>
</form>

  <div id="modal" uk-modal esc-close="false" bg-close="false">
    <div class="uk-modal-dialog uk-modal-body uk-modal-dialog-large" uk-overflow-auto >
      <div id="modal-text">
        <h2 class="uk-modal-title">Allele Search</h2>
        <div id="modal-gene-name"></div>
        <div>
          <span class="uk-text-bold">Region searched: </span>
          <div class="uk-margin-small-bottom">
            <span id="region-searched"></span>
            <span><span id="modal-search-region"></span></span>
          </div>
        </div>

          <div id="modal-body">
              <table id="variant-table" class="uk-table uk-table-divider uk-table-striped uk-table-small">
                  <thead style="background-color: lightgrey;">
                      <tr>
                          <th class="uk-text-emphasis">Chromosome</th>
                          <th class="uk-text-emphasis">Variant ID</th>
                          <th class="uk-text-emphasis">Position</th>
                          <th class="uk-text-emphasis">Ref</th>
                          <th class="uk-text-emphasis">Alt</th>
                      </tr>
                  </thead>
                  <tbody>
                      <tr id="data-row">
                          <td id="chromosome"></td>
                          <td id="id"></td>
                          <td id="position"></td>
                          <td id="ref"></td>
                          <td id="alt"></td>
                      </tr>
                      
                  </tbody>
                  <div id="loading-spinner" style="margin-left: 45%; display: block" uk-spinner ></div>
                  <tfoot>
                      <tr>
                          <td></td>
                      </tr>
                  </tfoot>
              </table>
          </div>
      </div>
        <div style="display: flex; justify-content: space-between;">
          <button class="uk-button uk-modal-close" type="button" onclick="refreshPage()">Close</button>
          <div>
            <button class="uk-button" onclick="downloadTxt()">Export txt</button> 
            <button class="uk-button" onclick="downloadCsv()">Export CSV</button> 
          </div>
        </div>
    </div>
  </div>
<br>
<br>

{% comment %} Traverse the annotation metadata, looking for a match to diversity_base {% endcomment %}
{% for metadata in site.data.datastore-metadata[genus][species].annotations %}
  {% for metadata_file in metadata[1] %}
    {% if metadata_file[0] contains "README" %}
        {% assign annot_coll_id = metadata_file[1].identifier %}
        {% assign identifier_parts = annot_coll_id | split: "." %}
        {% assign annot_base = identifier_parts | slice: 0, 2 | join: "." %}
        {% if annot_base == diversity_base %}
          <!-- <b>Annotation collection:</b> {{ annot_coll_id }} <br> -->
      {% endif %}
    {% endif %}
  {% endfor %}
{% endfor %} 


<script>
let variantCollection = document.getElementById("collection-name")
function onChange(){
  let value = JSON.parse(variantCollection.value)
  document.getElementById("synopsis").textContent = 
        `
        ${value.description}
        `
}
variantCollection.onchange = onChange
onChange()

function refreshPage() {
    window.location.reload();
}


/* ========================================================================= */
/* API calls */
/* ========================================================================= */
async function getData() {

  const geneName = document.getElementById("gene-name").value;
  let geneNames = geneName.split(",");
  let geneNamesString = "";

  geneNames.forEach(name => {
    geneNamesString += `"${name}",`;
  })

  geneNamesString = geneNamesString.slice(0, -1).replace(/\s/g, '');

  let myHeaders = new Headers();
  myHeaders.append("Content-Type", "text/plain");
  let searchString = `{"genes":[${geneNamesString}]}`;

  let requestOptions = {  
    method: 'POST',
    headers: myHeaders,
    body: searchString,
    redirect: 'follow'
  };

  await fetch("https://gcv.soybase.org/microservices/soybase/genes", requestOptions)
  .then(response =>{
      if(!response.ok){
          throw error
      }
      return response.json()})
  .then(result => {
    console.log("microservice",result)
    const chromosomeArray = result.genes[0].chromosome.split('.')
    const collectionName = `${chromosomeArray[1]}.${chromosomeArray[2]}`
    const chromosome = result.genes[0].chromosome
    const start = result.genes[0].fmin
    const stop = result.genes[0].fmax
    const geneName = result.genes[0].name

   document.getElementById("region-searched").textContent = `
   ${chromosome}:
   `
   document.getElementById("modal-gene-name").textContent = `
   ${geneName}
   `

      getDataFromFastA(chromosome,collectionName,start,stop, geneName);
  })
  .catch(error => console.log('error', error));

}

async function getDataFromFastA(chromosome,collectionName,start,stop,geneName) {
     let collectionVariantObj = JSON.parse(variantCollection.value)
     let collection = collectionVariantObj['collection-name'] //Dynamically get collection name from selections
     let fileName = collectionVariantObj.name
     let flankingRegion = document.getElementById("flank-region").value
     let startWithFlank
     let stopWithFlank
      if(flankingRegion){
        if(flankingRegion > parseInt(start)){
          startWithFlank = 0
          stopWithFlank = parseInt(stop)+parseInt(flankingRegion)
        }else{
          startWithFlank = parseInt(start)-parseInt(flankingRegion)
          stopWithFlank = parseInt(stop)+parseInt(flankingRegion)
        }
       
      }else{
        startWithFlank = start
        stopWithFlank = stop
      }

    let parsedGenomicRegion = document.getElementById("genomic-region").value.split('-')
    document.getElementById("genomic-region").value ? document.getElementById("modal-search-region").innerHTML = `<span>${parsedGenomicRegion[0]} - ${parsedGenomicRegion[1]}</span>` : document.getElementById("modal-search-region").innerHTML = `<span>${startWithFlank} - ${stopWithFlank}</span>`


    let urlRegion = `https://app.soybase.org/api/fasta-api/vcf/fetch/${chromosome}:${parsedGenomicRegion[0]}-${parsedGenomicRegion[1]}/https://data.legumeinfo.org/Glycine/max/diversity/${collection}/${fileName}`
    let urlFlank = `https://app.soybase.org/api/fasta-api/vcf/fetch/${chromosome}:${startWithFlank}-${stopWithFlank}/https://data.legumeinfo.org/Glycine/max/diversity/${collection}/${fileName}`
    let requestOptionsVariants = {
    method: 'GET',
    redirect: 'follow',
    headers: {
        "Accept": "application/json"
    }
    };

    let url = document.getElementById("genomic-region").value ? urlRegion : urlFlank

    await fetch(url, requestOptionsVariants)
    .then(response => response.json())
    .then(result => {
      console.log("fasta",result)
            const tableBody = document.getElementById("data-row").parentElement; 

            result.map((gene, index) => {
              const rowClass = index % 2 === 0 ? "even" : "odd"; 
              const tableRow = document.createElement("tr");
              tableRow.classList.add(rowClass);

              const chromosomeCell = document.createElement("td");
              const idCell = document.createElement("td");
              const posCell = document.createElement("td");
              const refCell = document.createElement("td");
              const altCell = document.createElement("td");

              chromosomeCell.textContent = gene.chrom
              idCell.textContent = gene.id;
              posCell.textContent = gene.pos;
              refCell.textContent = gene.ref;
              altCell.textContent = gene.alts;

              tableRow.appendChild(chromosomeCell)
              tableRow.appendChild(idCell);
              tableRow.appendChild(posCell);
              tableRow.appendChild(refCell);
              tableRow.appendChild(altCell);

              tableBody.appendChild(tableRow);
            });
            if(result){
              document.getElementById("loading-spinner").style.display="none"
              console.log(document.getElementById("modal").innerText)
            }
            let input = document.getElementById("modal-gene-name");
            input.value = '';
            
            })
            .catch(error => console.log('error', error))      
    }

function downloadTxt(){
  let content = document.getElementById("modal-text").innerText
  let filename = document.getElementById("modal-gene-name").innerText
  let a = document.createElement('a')
  let blob = new Blob([content], {type: 'text'})
  let url = URL.createObjectURL(blob)
  a.setAttribute('href', url)
  a.setAttribute('download', `${filename}.txt`)
  a.click()
} 

function downloadCsv(){
  let csv_data = [];
  let rows = document.getElementsByTagName('tr');
    for (let i = 0; i < rows.length; i++) {    
        let cols = rows[i].querySelectorAll('td,th');
        let csvrow = [];

        for (let j = 0; j < cols.length; j++) {
            csvrow.push(cols[j].innerHTML);
        }
        csv_data.push(csvrow.join(","));
    }
        csv_data = csv_data.join('\n');

        csvDownload(csv_data)
  }

function csvDownload(csv_data) {
  let filename = document.getElementById("modal-gene-name").innerText
  CSVFile = new Blob([csv_data], { type: "text/csv" });
  let temp_link = document.createElement('a');
  temp_link.download = `${filename}.csv`;
  let url = window.URL.createObjectURL(CSVFile);
  temp_link.href = url;
  temp_link.style.display = "none";
  document.body.appendChild(temp_link);
  temp_link.click();
  document.body.removeChild(temp_link);
  }
</script>