---
title: Allele search
sitemap: sequence map
tools_menu: true
---

<H2>Allele Search</H2>
<form>
  <label class="uk-form-label uk-text-default uk-text-bold" for="form-stacked-text">Step 1. Select a variant collection. Note that there are two reference assemblies to choose from.</label>
  <div>
    <select id="collection-name" class="uk-select uk-form-width-large uk-margin-small-top">
      {% assign genus = "Glycine" %}
      {% assign species = "max" %}
      {% for metadata in site.data.datastore-metadata[genus][species].diversity %}
        {% for metadata_file in metadata[1] %}
          {% if metadata_file[0] contains "MANIFEST" %}
            {% assign manifest = metadata_file %}
            {% for item in manifest[1] %}
              {% if item.applications contains "jbrowse" %}
                {% assign filename_parts = item.name | split: "." %}
                {% assign gensp = filename_parts[0] %}
                {% assign coll_parts = filename_parts | slice: 1, 4 %}
                {% assign diversity_base = filename_parts | slice: 1, 2 | join: "." %}
                {% capture collection_name %}{{ coll_parts | join: "." }}{% endcapture %}
                {%- capture collection_item -%}
                {
                  "name": "{{item.name}}",
                  "collection-name": "{{collection_name}}",
                  "description": "{{item.description}}"
                }
                {%- endcapture %}
                <option value='{{- collection_item | strip_newlines | strip_html -}}'>{{collection_name}}</option>
              {% endif %}
            {% endfor %}
          {% endif %}
        {% endfor %}
      {% endfor %}
    </select>
  </div>
  <div style="margin-left: 10px;">
    <label class="uk-form-label uk-text-default uk-text-emphasis">Synopsis: <span id="synopsis"></span></label>
  </div>

  <div class="uk-margin" style="display: flex; justify-content: start;">
    <div>
      <div>
        <label class="uk-form-label uk-text-default uk-text-bold" for="form-stacked-text">Step 2. Enter gene ID and size of left and right flanking regions...</label>
        <div><input class="uk-input uk-form-width-large uk-margin-small-top" type="text" id="gene-name" placeholder="Enter gene name"></input></div>
      </div>
      <div>
        <label class="uk-form-label uk-text-default uk-text-muted">e.g glyma.Wm82.gnm4.ann1.Glyma.16G044100</label>
      </div>
    </div>
    <div >
      <br>
      <div class="uk-margin-small-left">
        <input class="uk-input uk-form-width-medium uk-margin-small-top" type="text" id="flank-region"  placeholder="Enter flanking region">
      </input>
      <div>
        <label class="uk-form-label uk-text-default uk-text-muted">e.g 50000</label>
      </div>
    </div>
    </div>
  </div>
  
  <div class="uk-margin">
    <div>
        <label class="uk-form-label uk-text-default uk-text-bold" for="form-stacked-text">... or a genomic region, as a chromosome: start-end.</label>
    </div>
  <div>
    <input class="uk-input uk-form-width-medium uk-margin-small-top" type="text" id="genomic-region" rows="10" cols="30" placeholder="Start-End"></input>
    <div>
      <div><label class="uk-form-label uk-text-default uk-text-muted">e.g 1000000-200000</label></div>
    </div>
  </div>
    <input class="uk-button uk-margin" uk-toggle="target: #modal" id="submit" type="button" value="Search" onclick="getData()" style="margin-top: 10px;"/>
  </div>
</form>

  <div id="modal" uk-modal bg-close="false">
    <div class="uk-modal-dialog uk-modal-body">
        <h2 class="uk-modal-title">Allele Search</h2>
            <div id="modal-body">
                <table id="variant-table" class="uk-table uk-table-divider uk-table-striped uk-table-small">
                    <caption>Search Results for <span id="modal-title"></span></caption>
                    <caption>Genomic Region: <span id="modal-search-region"></span></caption>
                    <thead>
                        <tr>
                            <th>ID</th>
                            <th>Position</th>
                            <th>Alleles</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr id="data-row">
                            <td id="id"></td>
                            <td id="position"></td>
                            <td id="alleles"></td>
                        </tr>
                    </tbody>
                    <tfoot>
                        <tr>
                            <td></td>
                        </tr>
                    </tfoot>
                </table>
            </div>
        <button class="uk-button uk-modal-close" type="button" onclick="deleteRow()">Close</button>
    </div>
  </div>
<!-- <br>
<br>
{% for metadata in site.data.datastore-metadata[genus][species].diversity %}
  {% for metadata_file in metadata[1] %}
    {% if metadata_file[0] contains "MANIFEST" %}
      {% assign manifest = metadata_file %}
      {% for item in manifest[1] %}
        {% if item.applications contains "jbrowse" %}
          {% assign filename_parts = item.name | split: "." %}
          {% assign gensp = filename_parts[0] %}
          {% assign coll_parts = filename_parts | slice: 1, 4 %}
          {% assign diversity_base = filename_parts | slice: 1, 2 | join: "." %}
          {% capture collection_name %}{{ coll_parts | join: "." }}{% endcapture %}
          <b>VCF file:</b> {{ item.name }} <br>
          <b>Collection:</b> {{ collection_name }} <br>
          <b>Description:</b> {{ item.description }} <br>
          
          <br>
        {% endif %}
      {% endfor %}
    {% endif %}
  {% endfor %}
{% endfor %}
-->

{% comment %} Traverse the annotation metadata, looking for a match to diversity_base {% endcomment %}
{% for metadata in site.data.datastore-metadata[genus][species].annotations %}
  {% for metadata_file in metadata[1] %}
    {% if metadata_file[0] contains "README" %}
        {% assign annot_coll_id = metadata_file[1].identifier %}
        {% assign identifier_parts = annot_coll_id | split: "." %}
        {% assign annot_base = identifier_parts | slice: 0, 2 | join: "." %}
        {% if annot_base == diversity_base %}
          <b>Annotation collection:</b> {{ annot_coll_id }} <br>
      {% endif %}
    {% endif %}
  {% endfor %}
{% endfor %} 


<script>
let variantCollection = document.getElementById("collection-name")
function onChange(){
  let value = JSON.parse(variantCollection.value)
  console.log("value",value)
  document.getElementById("synopsis").innerHTML = 
        `
        <span>${value.description}</span>
        `
}
variantCollection.onchange = onChange
onChange()

function deleteRow() {
    window.location.reload();
}


/* ========================================================================= */
/* API calls */
/* ========================================================================= */
async function getData() {

const geneName = document.getElementById("gene-name").value;
let geneNames = geneName.split(",");
let geneNamesString = "";

geneNames.forEach(name => {
  geneNamesString += `"${name}",`;
})

geneNamesString = geneNamesString.slice(0, -1).replace(/\s/g, '');

let myHeaders = new Headers();
myHeaders.append("Content-Type", "text/plain");
let searchString = `{"genes":[${geneNamesString}]}`;

let requestOptions = {
  method: 'POST',
  headers: myHeaders,
  body: searchString,
  redirect: 'follow'
};

await fetch("https://gcv.soybase.org/microservices/soybase/genes", requestOptions)
  .then(response =>{
      if(!response.ok){
          throw error
      }
      return response.json()})
  .then(result => {
    let chromosomeArray = result.genes[0].chromosome.split('.')
    let collectionName = `${chromosomeArray[1]}.${chromosomeArray[2]}`
    let chromosome = result.genes[0].chromosome
    let start = result.genes[0].fmin
    let stop = result.genes[0].fmax

    console.log(result)
   document.getElementById("modal-title").innerHTML = `
   <span uk-tooltip=${result.genes[0].chromosome}>${result.genes[0].name}</span>
   `

      getDataFromFastA(chromosome,collectionName,start,stop);
  })
  .catch(error => console.log('error', error));


  function getDataFromFastA(chromosome,collectionName,start,stop) {
       console.log("CollectionName",collectionName)
       let collectionVariantObj = JSON.parse(variantCollection.value)
       console.log("line226",collectionVariantObj['collection-name'])
       let collection = collectionVariantObj['collection-name'] //Dynamically get collection name from selections
       let fileName = collectionVariantObj.name
       console.log("collect",collection)

       let flankingRegion = document.getElementById("flank-region").value
       console.log("Flank",flankingRegion)
       let startWithFlank
       let stopWithFlank
        if(flankingRegion){
          console.log(true)
          if(flankingRegion > parseInt(start)){
            startWithFlank = 0
            stopWithFlank = parseInt(stop)+parseInt(flankingRegion)
          }else{
            startWithFlank = parseInt(start)-parseInt(flankingRegion)
            stopWithFlank = parseInt(stop)+parseInt(flankingRegion)
          }
         
        }else{
          console.log(false)
          startWithFlank = start
          stopWithFlank = stop
        }

      let parsedGenomicRegion = document.getElementById("genomic-region").value.split('-')
      document.getElementById("genomic-region").value ? document.getElementById("modal-search-region").innerHTML = `<span>${parsedGenomicRegion[0]} - ${parsedGenomicRegion[1]}</span>` : document.getElementById("modal-search-region").innerHTML = `<span>${startWithFlank} - ${stopWithFlank}</span>`


      let urlRegion = `https://app.soybase.org/api/fasta-api/vcf/fetch/${chromosome}:${parsedGenomicRegion[0]}-${parsedGenomicRegion[1]}/https://data.legumeinfo.org/Glycine/max/diversity/${collection}/${fileName}`
      let urlFlank = `https://app.soybase.org/api/fasta-api/vcf/fetch/${chromosome}:${startWithFlank}-${stopWithFlank}/https://data.legumeinfo.org/Glycine/max/diversity/${collection}/${fileName}`
      let requestOptionsVariants = {
      method: 'GET',
      redirect: 'follow',
      headers: {
          "Accept": "application/json"
      }
      };

      let url = document.getElementById("genomic-region").value ? urlRegion : urlFlank

      fetch(url, requestOptionsVariants)
      .then(response => response.json())
      .then(result => {
              const tableBody = document.getElementById("data-row").parentElement; 

              result.map((gene, index) => {
                const rowClass = index % 2 === 0 ? "even" : "odd"; 
                const tableRow = document.createElement("tr");
                tableRow.classList.add(rowClass);

                const idCell = document.createElement("td");
                const posCell = document.createElement("td");
                const allelesCell = document.createElement("td");

                idCell.textContent = gene.id;
                posCell.textContent = gene.pos;
                allelesCell.textContent = gene.alleles;

                tableRow.appendChild(idCell);
                tableRow.appendChild(posCell);
                tableRow.appendChild(allelesCell);

                tableBody.appendChild(tableRow);
              });
              let input = document.getElementById("gene-name");
              input.value = '';
              
              })
              .catch(error => console.log('error', error))      
      }
  }
</script>