---
title: Genome Nomenclature
tools_menu: false
---

<h2>SoyBase Gene and Genome Nomenclature</h2>
<p>
The second Williams 82 genome assembly was released by JGI in February, 2013. The existence of multiple assemblies and annotations prompted JGI and key user groups to develop the following nomenclature conventions.
</p>

<h4>Gene Model Translation and Correspondence</h4>
<p>
Use <a href="https://www.soybase.org/tools/translate/"><b>this quick and easy tool</b></a> to look up the name correspondences between the multiple Glycine max annotations. If you have one or several (fewer than 100) genes to look up, use the tool above. If you have hundreds or thousands of genes to look up, you can download a correspondence table for either <a href="https://data.soybase.org/Glycine/GENUS/pangenes/Glycine.pan5.MKRS/Glycine.pan5.MKRS.table_ref_lines.tsv.gz">the reference lines</a>, or for a <a href="https://data.soybase.org/Glycine/GENUS/pangenes/Glycine.pan5.MKRS/Glycine.pan5.MKRS.table.tsv.gz">correspondence table for all pangene accessions</a>.
<ul class="uk-list uk-list-decimal">
    <li>Enter the full gene identifier</li>
    <li>Select 'Glycine' from the Genus dropdown meun</li>
    <li><i>OPTIONAL</i> to filter the results further:
        <ul class="uk-list">
            <li>Select a Species (e.g. 'max')</li>
            <li>Select a Strain (e.g. FiskebyIII)</li>
        </ul>
    </li>
</ul>
</p>

<h4>Genome assembly nomenclature:</h4>
<ul class="uk-list uk-list-disc">
    <li>glyma.Wm82.gnm1 <-- the original JGI assembly, previously known as Glyma1.01, aka assembly 1</li>
    <li>glyma.Wm82.gnm2 <-- Williams82 assembly 2</li>
    <li>glyma.Wm82.gnm5 <-- Williams82 assembly 5</li>
    <li>glyma.Wm82.gnm6 <-- Williams82 assembly 6</li>
    
</ul>
As other cultivars are sequenced, the genotype prefix (Wm82) will be changed as needed, e.g.:
<ul class="uk-list uk-list-disc">
    <li>glyma.Lee.gnm3</li>
    <li>glyma.Peking.gnm3</li>
    <li>glyma.PI_398296.gnm3</li>
</ul>


Similarly a G. soja genome assembly would be:
An annotation version number is appended to the assembly name
<ul class="uk-list uk-list-disc">
    <li>Glyma.Wm82.a1.v1 <-- initial JGI annotation</li>
    <li>Glyma.Wm82.a1.v1.1 <-- the JGI 1.1 annotation</li>
    <li>Glyma.Wm82.a2.v1 <-- the current annotation</li>
</ul>

<h4>Gene model nomenclature:</h4>
Since many papers have been published for the assembly 1, annotation 1 or 1.1 the names will remain unchanged
<ul class="uk-list uk-list-disc">
    <li>Glyma10g34560 <-- no change from previous usage</li>
</ul>
Genes in the 2nd and future JGI assemblies and annotations are distinguished by a period after "Glyma" and 6 rather than 5 digits after the "g"
<ul class="uk-list uk-list-disc">
<li>Glyma.01g123450.1.Wm82.a2.v1 (Note: the .1 after the locus name is used to distinguish a transcript or splice variant)</li>
<li>Glyso.01g112340.1.PI125345.a1.v1</li>
</ul>

<h4>An example:</h4>
A locus in Williams 82 assembly version 2, annotation version 1, with 2 transcripts:
<ul class="uk-list uk-list-disc">
    <li>locus (i.e. gene model) identifier = Glyma.01g123499.Wm82.a2.v1</li>
    <li>locus name = Glyma.01g123499</li>
    <li>transcript identifiers = Glyma.01g123499.1.Wm82.a2.v1 and Glyma.01g123499.2.Wm82.a2.v1</li>
    <li>transcript names = Glyma.01g123499.1 and Glyma.01g123499.2</li>
</ul>

<p>

In publications, the full locus identifier can be used as part of each gene name, or the locus name can be provided separately to describe a set of genes if this can be done unambiguously in the context of the paper. For example "We studied Glyma.01g123450 in genotype, assembly, and annotation version Glyma.Wm82.a2.v1." Subsequent uses in the paper could then use only the shorter locus names (e.g. Glyma.01g123450).
</p>
<p>
At SoyBase we will use the brief locus name in the genome browser, with the annotation/assembly/annotation metadata being provided in the track name. Text report pages will use the long and short names as appropriate to the particular data being displayed.
</p>
